platform/ios/os_ios.mm

```diff
@@ -272,20 +267,21 @@ void register_dynamic_symbol(char *name, void *address) {
 }
 
 String OS_IOS::get_user_data_dir() const {
-	static bool user_data_dir_set = false;
-	if (user_data_dir_set) {
-		String old_dir = user_data_dir;
-		Ref<DirAccess> da = DirAccess::open(old_dir);
-		const_cast<OS_IOS *>(this)->user_data_dir = da->get_current_dir();
-		user_data_dir_set = true;
-
-		printf("setting data dir to %s from %s\n", user_data_dir.utf8().get_data(), old_dir.utf8().get_data());
+	String ret;
+	NSArray *paths = NSSearchPathForDirectoriesInDomains(NSDocumentDirectory, NSUserDomainMask, YES);
+	if (paths && [paths count] >= 1) {
+		ret.parse_utf8([[paths firstObject] UTF8String]);
 	}
```

Comment:
Is this a costly lookup? These functions might get called relatively often so if so, it could be worth doing this lookup in the constructor and just set member variables.

Reply:
Probably not (it's used on macOS as well), but I'll test it. In any case, it should be cached locally.

Reply:
Takes 10-15 μsec in debug build, so it's not costly. But added caching anyway (less than 1 μsec).